plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from 'winston';
+import {
+  ClusterDetails,
+  KubernetesFetcher,
+  KubernetesObjectTypes,
+  KubernetesRequestBody,
+  KubernetesServiceLocator,
+} from '../types/types';
+import { KubernetesAuthTranslator } from '../kubernetes-auth-translator/types';
+import { KubernetesAuthTranslatorGenerator } from '../kubernetes-auth-translator/KubernetesAuthTranslatorGenerator';
+
+const DEFAULT_OBJECTS = new Set<KubernetesObjectTypes>([
+  'pods',
+  'services',
+  'configmaps',
+  'deployments',
+  'replicasets',
+  'horizontalpodautoscalers',
+  'ingresses',
+]);
+
+export class KubernetesFanOutHandler {
+  private readonly logger: Logger;
+  private readonly fetcher: KubernetesFetcher;
+  private readonly serviceLocator: KubernetesServiceLocator;
+
+  constructor(
+    logger: Logger,
+    fetcher: KubernetesFetcher,
+    serviceLocator: KubernetesServiceLocator,
+  ) {
+    this.logger = logger;
+    this.fetcher = fetcher;
+    this.serviceLocator = serviceLocator;
+  }
+
+  async getKubernetesObjectsByEntity(
+    requestBody: KubernetesRequestBody,
+    objectTypesToFetch: Set<KubernetesObjectTypes> = DEFAULT_OBJECTS,
+  ) {
+    const entityName = requestBody.entity.metadata.name;
+
+    const clusterDetails: ClusterDetails[] = await this.serviceLocator.getClustersByServiceId(
+      entityName,
```

Comment:
Is the locator a per-namespace thing, or are namespaces simply left as a future extension? Big caveat that I may be misunderstanding things :)

Reply:
Namespaces are for the future, the serviceLocator is an interface to map a service to which cluster it actually lives on, but ATM the only implementation assumes that every service lives on every cluster